### PR TITLE
core: Update sounds in rootless SWF (close #11666)

### LINKED ISSUE #11666: Super Crazy Guitar maniac Deluxe 4: level doesn't end at the end of the song
### Describe the bug

SWF URL: http://fgn.cdn.serverable.com/common/flashes/ff/s/super_crazy_guitar_maniac_deluxe_4.swf?d=20120216203039

steps to reproduce: start the game, choose any song, wait for the song to end. the level stays without any more arrows, with whatever animation that was playing in the background by the end.

### Expected behavior

the level ends after the song ends (the exact duration specified during level selection), the score is displayed, and a menu given to choose the next song.

### Affected platform

Desktop app

### Operating system

Linux (Fedora Silverblue 38)

### Browser

_No response_

### Additional information

command line used: `ruffle -g gl super_crazy_guitar_maniac_deluxe_4.swf`

using `GeForce GTX 760` with `nouveau` driver.

ruffle nightly: `b3c40ecda107ee33cbc44868fcd7dfdbdc634775`

EDIT: reproduced in https://ruffle.rs/demo/ on `firefox 114.0.2  (flatpak)`

### PR DESCRIPTION
**Pull request by nivkner**
the current implementation assumes a root clip exists, despite that not being required:
https://github.com/ruffle-rs/ruffle/blob/f24aabf74efbb39ec3858cfdbbe24d33c9dae25b/core/src/display_object/stage.rs#L655

unlike avm2, the avm1 event dispatch is tied to a clip, so that is skipped if there is no root (i don't know how to choose the correct clip in this case).

the position is changed to the duration though, since the sound did finish, regardless of what events are dispatched.

i used apache flex to build the SWF. however, ruffle doesn't currently support `Sound(url)` even if the URL is to a local file, and according to [this answer](https://stackoverflow.com/questions/4068237/how-to-use-flash-filesystem-filestream/4073924) apache flex does not support `flash.filesystem.FileStream` so i can't use use `loadCompressedDataFromByteArray` instead.
hence the test skip.

EDIT: removed the test skip, fixes #11666

### CODE REVIEW

File: tests/tests/swfs/avm2/sound_rootless/test.toml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+num_frames = 1
+ignore = true # enable once flash.media.Sound can load from file:// URL
```
Comment (Lord-McSweeney) [inline tests/tests/swfs/avm2/sound_rootless/test.toml:2]: AFAIK You can fix this by replacing this:
```

            var sound_url:URLRequest = new URLRequest("file://noise.mp3");
            var sound:Sound = new Sound(sound_url);
```
with this:
```

            var sound_url:URLRequest = new URLRequest("noise.mp3");
            var sound:Sound = new Sound();
            sound.load(sound_url);
```

Comment (nivkner) [reply tests/tests/swfs/avm2/sound_rootless/test.toml:2]: changed the test as described. didn't play the sound/
for some reason i am getting the following when running with `RUST_LOG=warn,ruffle_core=debug,avm_trace=trace`:
```
2023-06-24T22:30:20.176283Z  INFO run_frame:run_all_phases_avm2: avm_trace: Initial state
2023-06-24T22:30:20.176327Z  INFO run_frame:run_all_phases_avm2: avm_trace: Removed root
2023-06-24T22:30:20.176389Z  WARN run_frame:run_all_phases_avm2: ruffle_core::stub: Encountered stub: AVM2 flash.media.Sound constructor with arguments
2023-06-24T22:30:20.176415Z  WARN run_frame:run_all_phases_avm2: ruffle_core::stub: Encountered stub: AVM2 flash.media.Sound.load() with context
2023-06-24T22:30:20.176437Z  INFO run_frame:run_all_phases_avm2: avm_trace: Loaded sound
2023-06-24T22:30:20.176493Z  INFO run_frame:run_all_phases_avm2: avm_trace: Playing sound
2023-06-24T22:30:20.176505Z  INFO run_frame:run_all_phases_avm2: avm_trace: [object SoundChannel]
```
despite the fact that `Sound` isn't getting an argument and `load` isn't getting a context

Comment (Lord-McSweeney) [reply tests/tests/swfs/avm2/sound_rootless/test.toml:2]: `Encountered stub: AVM2 flash.media.Sound constructor with arguments`
You did something else wrong, in that case. Can you commit your changes so I can review?

Comment (nivkner) [reply tests/tests/swfs/avm2/sound_rootless/test.toml:2]: sent the commit with the above changes

File: tests/tests/swfs/avm2/sound_rootless/Test.as [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        public function Test() {
+    		    actualStage = stage;
+
+            trace("Initial state");
+
+    			  stage.removeChildAt(0);
+
+    			  trace("Removed root");
+
+            var sound_url:URLRequest = new URLRequest("file://noise.mp3");
```
Comment (Lord-McSweeney) [inline tests/tests/swfs/avm2/sound_rootless/Test.as:21]: This is incorrect, it should be `new URLRequest("noise.mp3");`

File: tests/tests/swfs/avm2/sound_rootless/test.toml (hunk 1 above)
Comment (Lord-McSweeney) [reply tests/tests/swfs/avm2/sound_rootless/test.toml:2]: Your SWF file does not match up with your AS code

File: tests/tests/swfs/avm2/sound_rootless/Test.as (hunk 2 above)
Comment (nivkner) [reply tests/tests/swfs/avm2/sound_rootless/Test.as]: changed `"file://noise.mp3"` to `"noise.mp3"`

File: tests/tests/swfs/avm2/sound_rootless/test.toml (hunk 1 above)
Comment (nivkner) [reply tests/tests/swfs/avm2/sound_rootless/test.toml]: removed the test ignore

**Review by Lord-McSweeney — APPROVED** [review]

**Review by adrian17 — APPROVED** [review]

**Review by Dinnerbone — APPROVED** [review]

**Review by Aaron1011 — APPROVED** [review]

**Pull request merged**